hooks/validators/deny_host_path_volumes.go

```diff
@@ -2,23 +2,45 @@ package validators
 
 import (
 	"context"
+	"path/filepath"
+	"strings"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
+type AllowedHostPath struct {
+	PathPrefix string `json:"pathPrefix"`
+}
+
 // DenyHostPathVolumes is a Validator that denies usage of HostPath volumes
-type DenyHostPathVolumes struct{}
+type DenyHostPathVolumes struct {
+	allowedHostPaths []AllowedHostPath
+}
+
+func NewDenyHostPaths(hostPaths []AllowedHostPath) DenyHostPathVolumes {
+	return DenyHostPathVolumes{allowedHostPaths: hostPaths}
+}
 
 func (v DenyHostPathVolumes) Validate(ctx context.Context, pod *corev1.Pod) field.ErrorList {
 	p := field.NewPath("spec").Child("volumes")
 	var errs field.ErrorList
+
+volume_for:
 	for i, vol := range pod.Spec.Volumes {
 		if vol.HostPath == nil {
 			continue
 		}
-		if len(vol.HostPath.Path) != 0 || vol.HostPath.Type != nil {
-			errs = append(errs, field.Forbidden(p.Index(i), "Volume type HostPath is not allowed to be used"))
+		if v.allowedHostPaths == nil {
```

Comment:
Use `len(v.allowsHostPaths) == 0` to cover an empty slice.

This `if` is redundant because `for _, allowed := range v.allowedHostPaths` works correctly for `nil`.